### PR TITLE
Feature/ws origin env variables

### LINKED ISSUE #8502: Support environment variable in addition to --allow-websocket-origin
Hey folks! 

This is more of a feature request

I'm running bokeh in docker in multiple environment and being able to specify an environment variable `BOKEH_ALLOWED_WS_ORIGIN` with a list of domains instead of using `--allow-websocket-origin` would make my life easier. 

Docker does not support environment variables in `CMD` and that would force me to write a small wrapper script to run bokeh, which is not ideal.

I can work on a PR if you think that's something worth supporting. I wanted to get your opinions on this first.

### PR DESCRIPTION
**Pull request by kuuji**
All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [x] issues: fixes #8502
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

I made the choice to not override the default flag and just merge the environment value with the flag.

That's because IMO the cli flag should override environment variables, and the way it is right now, there's always at least a default flag so it would always override.

Other than that, the rest is pretty straightforward. Added few tests to cover the new environment variable and updated the  environment variable doc.

### CODE REVIEW

File: bokeh/settings.py [hunk 1]
```diff
@@ -8,6 +8,13 @@
 ''' Control global configuration options with environment variables.
 A global settings object that other parts of Bokeh can refer to.
 
+``BOKEH_ALLOW_WS_ORIGIN`` --- List of websocket origins allowed to access bokeh.
+
+  Comma separated list of domains that need to access bokeh websocket interface.
+  This can also be provided using the --allow-websocket-origin parameter. 
+
+  Note: No option overrides the other. We instead merge the list from --allow-websocket-origin
+  and the list from BOKEH_ALLOW_WS_ORIGIN
```
Comment (bryevdv) [inline bokeh/settings.py]: I think this merits some discussion. My experience is that "standard" semantics for this sort of thing is for the env var to completely override other settings. I think merging might be surprising to users.

File: bokeh/server/views/ws.py [hunk 2]
```diff
@@ -65,14 +66,16 @@ def check_origin(self, origin):
...
+            allowed_hosts |= set(settings.allowed_ws_origin())
 
         allowed = check_whitelist(origin_host, allowed_hosts)
         if allowed:
             return True
         else:
             log.error("Refusing websocket connection from Origin '%s'; \
-                      use --allow-websocket-origin=%s to permit this; currently we allow origins %r",
-                      origin, origin_host, allowed_hosts)
+                      use --allow-websocket-origin=%s or BOKEH_ALLOW_WS_ORIGIN=%s to permit this; currently we allow origins %r",
```
Comment (bryevdv) [inline bokeh/server/views/ws.py]: "or *set* BOKEH_ALLOW_WS_ORIGIN=" is slightly more explicit

**Comment by kuuji** [comment]
No problem @bryevdv 👍 . Thanks for the awesome work on bokeh!

I made some adjustments following your comments ;)

**Comment by bryevdv** [comment]
@kuuji Thank you for the kind words. Last comment is just that I don't see a test that explicitly exercises the "listable" nature of the env var. If it is not too difficult I would suggest adding something. Could either be a direct test of `_get_list` in `bokeh.settings` tests or could be a test with a list value for the env var in the server tests.

**Comment by bryevdv** [comment]
@kuuji Thanks for adding them, and my apologies I did not realize there was not a test module yet. It's really good to get one started in that case. Since I didn't know there was not a module, I did not mention using our standard module boilerplate. Since trying to explain what goes where seemed like it might result in alot of tedious back and forth for you, I went ahead and just edited the new test file directly to save effort. I expect there may be some issues on TravisCI since I just edited from the GH web UI, but they should be simple to clear up once the build finishes.

**Comment by bryevdv** [comment]
@kuuji OK I fixed up the issues with applying the boilerplate. All that remains are some linter/docs failures. The docs error is:

> sphinx.errors.SphinxWarning: /home/travis/build/bokeh/bokeh/sphinx/source/docs/dev_guide/env_vars.rst:9:Title underline too short.

Sphinx is picky, section underlines have to match exactly the length of the text above. And the linter issues are:

> E         File contains trailing whitespace: bokeh/command/subcommands/serve.py, line 125.
> E         File contains trailing whitespace: bokeh/settings.py, line 14.
> E         File contains trailing whitespace: bokeh/settings.py, line 217.
> E         File contains trailing whitespace: sphinx/source/docs/dev_guide/env_vars.rst, line 12.

Once those are fixed the build should go green and we can get this merged!

**Comment by kuuji** [comment]
Sorry about that @bryevdv. 
I didn't realized the linter was actually a test and I think the way that I ran the tests locally did not run that linter test.

I stole the command from the CI job and looks like everything is good now with my last commit. I was also able to build the sphinx doc.

And thanks for adjusting the test class. Is the boilerplate documented anywhere?

**Comment by bryevdv** [comment]
Thanks for the PR @kuuji !

> Is the boilerplate documented anywhere?

We are still in the middle of applying it across the entire codebase. There is some discussion at https://github.com/bokeh/bokeh/issues/8207 It should go in the Dev Guide at some point

**Pull request merged**